geodetic_utils/include/geodetic_utils/geodetic_conv.h

```diff
@@ -0,0 +1,217 @@
+  {
+    // Convert geodetic coordinates to ECEF.
+    // http://code.google.com/p/pysatel/source/browse/trunk/coord.py?r=22
+    double lat_rad = deg2Rad(lat);
+    double lon_rad = deg2Rad(lon);
+    double xi = sqrt(1 - esq * sin(lat_rad) * sin(lat_rad));
+    x = (a / xi + altitude) * cos(lat_rad) * cos(lon_rad);
+    y = (a / xi + altitude) * cos(lat_rad) * sin(lon_rad);
+    z = (a / xi * (1 - esq) + altitude) * sin(lat_rad);
+  }
+
+  void ecef2Geodetic(const double x, const double y, const double z, double& lat, double& lon,
+                     double& altitude)
+  {
+    // Convert ECEF coordinates to geodetic coordinates.
+    // J. Zhu, "Conversion of Earth-centered Earth-fixed coordinates
+    // to geodetic coordinates," IEEE Transactions on Aerospace and
+    // Electronic Systems, vol. 30, pp. 957-961, 1994.
+
+    double r = sqrt(x * x + y * y);
+    double Esq = a * a - b * b;
+    double F = 54 * b * b * z * z;
+    double G = r * r + (1 - esq) * z * z - esq * Esq;
+    double C = (esq * esq * F * r * r) / pow(G, 3);
+    double S = cbrt(1 + C + sqrt(C * C + 2 * C));
+    double P = F / (3 * pow((S + 1 / S + 1), 2) * G * G);
+    double Q = sqrt(1 + 2 * esq * esq * P);
+    double r_0 = -(P * esq * r) / (1 + Q)
+        + sqrt(
+            0.5 * a * a * (1 + 1.0 / Q) - P * (1 - esq) * z * z / (Q * (1 + Q)) - 0.5 * P * r * r);
+    double U = sqrt(pow((r - esq * r_0), 2) + z * z);
+    double V = sqrt(pow((r - esq * r_0), 2) + (1 - esq) * z * z);
+    double Z_0 = b * b * z / (a * V);
+    altitude = U * (1 - b * b / (a * V));
+    lat = rad2Deg(atan((z + e1sq * Z_0) / r));
+    lon = rad2Deg(atan2(y, x));
+  }
+
+  void ecef2Ned(const double x, const double y, const double z, double& north, double& east,
+                double& depth)
+  {
+    // Converts ECEF coordinate position into local-tangent-plane NED.
+    // Coordinates relative to given ECEF coordinate frame.
+
+    Eigen::Vector3d vect, ret;
+    vect(0) = x - _init_ecef_x;
+    vect(1) = y - _init_ecef_y;
+    vect(2) = z - _init_ecef_z;
+    ret = _ecef_to_ned_matrix * vect;
+    north = ret(0);
+    east = ret(1);
+    depth = -ret(2);
+  }
+
+  void ned2Ecef(const double north, const double east, const double depth, double& x, double& y,
+                double& z)
+  {
+    // NED (north/east/down) to ECEF coordinates
+    Eigen::Vector3d ned, ret;
+    ned(0) = north;
+    ned(1) = east;
+    ned(2) = -depth;
+    ret = _ned_to_ecef_matrix * ned;
+    x = ret(0) + _init_ecef_x;
+    y = ret(1) + _init_ecef_y;
+    z = ret(2) + _init_ecef_z;
+  }
+
+  void geodetic2Ned(const double lat, const double lon, const double altitude, double& north,
+                    double& east, double& depth)
+  {
+    // Geodetic position to local NED frame
+    double x, y, z;
+    geodetic2Ecef(lat, lon, altitude, x, y, z);
+    ecef2Ned(x, y, z, north, east, depth);
+  }
+
+  void ned2Geodetic(const double north, const double east, const double depth, double& lat,
```

Comment:
`d` usually refers to `down` not `depth` https://en.wikipedia.org/wiki/North_east_down